### PR TITLE
Fix 403 apk v3 signatures

### PR DESCRIPTION
**Pull request by g-k**
fix: #403 

refs: https://jira.mozilla.com/browse/RELENG-9

Functional Tests:

- [x] force verify test in `apk2_test.go` to fail artificially and confirm verify output includes v3 format verification

```
=== CONT  TestSignFile/VerifyWithAPKSigner                                                                                                   
    apk2_test.go:126: error verifying apk signature: %!s(<nil>)                                                                              
        Verifies                                                                                                                             
        Verified using v1 scheme (JAR signing): true                                                                                         
        Verified using v2 scheme (APK Signature Scheme v2): true                                                                             
        Verified using v3 scheme (APK Signature Scheme v3): true                                                                             
        Number of signers: 1   
```  

r? @ajvb

NB: I'm going to hold off on landing this until I get the go ahead from RelEng and Mobile

### CODE REVIEW

File: signer/apk2/apk2.go [hunk 1]
```diff
@@ -119,11 +119,12 @@ func (s *APK2Signer) SignFile(file []byte, options interface{}) (signer.SignedFi
 	defer os.Remove(tmpAPKFile.Name())
 	ioutil.WriteFile(tmpAPKFile.Name(), file, 0755)
 
-	apkSigCmd := exec.Command("java", "-jar", "/usr/bin/apksigner", "sign",
```
Comment (g-k) [inline signer/apk2/apk2.go:122]: `/usr/bin/apksigner` is a shell script. No idea why we were calling it as a jar file before.

**Review by ajvb — APPROVED** [review]
lgtm

File: signer/apk2/apk2_test.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 func assertNewSignerWithConfOK(t *testing.T, conf signer.Configuration) *APK2Signer {
-	s, err := New(apk2signerconf)
```
Comment (g-k) [inline signer/apk2/apk2_test.go:12]: This was a bug with the old test code.

File: signer/apk2/apk2.go [hunk 3]
```diff
@@ -119,14 +137,25 @@ func (s *APK2Signer) SignFile(file []byte, options interface{}) (signer.SignedFi
...
+	args := []string{
+		"-jar", "/usr/share/java/apksigner.jar", "sign",
 		"--v1-signing-enabled", "true",
 		"--v2-signing-enabled", "true",
+	}
+	if s.v3Enabled {
+		args = append(args, "--v3-signing-enabled", "true")
+	} else {
+		// apksigner signs with v3 if the minsdk version supports it
+		args = append(args, "--v3-signing-enabled", "false")
```
Comment (g-k) [inline signer/apk2/apk2.go:149]: Have to explicitly disable v3 signing after upgrading `apksigner`

**Comment by g-k** [comment]
Got feedback from Aki that the mobile team would like to enable v3 sigs gradually, so autograph needs to support signing with and without v3 sigs.

I added a `v3enabled` mode to the `apk2` signer and integration tests.

I didn't add unit tests, since the signer doesn't have them and it'd require a larger change to have it take a mocked or dummy filesystem or `ioutil` implementation.

r? @ajvb

**Review by ajvb — APPROVED** [review]
Looks great

**Comment by g-k** [comment]
Thanks AJ!

**Pull request merged**